pkg/controller/tas/resource_flavor.go

```diff
@@ -236,7 +236,19 @@ func nodeBelongsToFlavor(node *corev1.Node, nodeLabels map[string]string, levels
 	return true
 }
 
+// normalizeNode zeros out fields that should be ignored during comparison.
+func normalizeNode(node *corev1.Node) {
+	// Ignore ResourceVersion (metadata)
+	node.ObjectMeta.ResourceVersion = ""
+
+	// ManagedFields often contain internal timestamps and high-churn metadata
+	node.ObjectMeta.ManagedFields = nil
+}
+
 // checkNodeSchedulingPropertiesChanged checks if the node update affects TAS scheduling.
 func checkNodeSchedulingPropertiesChanged(oldNode, newNode *corev1.Node) bool {
-	return !nodeSemantic.DeepEqual(oldNode, newNode)
+	o, n := oldNode.DeepCopy(), newNode.DeepCopy()
```

Comment:
nit, this requires DeepCopying both versions of the node. I think it would be better and more consistent with the current code to just implement it inside `nodeSemantic`. We could have there nodeMetadataEqual, say:

```
func nodeMetadataEqual(a, b metav1.ObjectMeta) bool {
	aCopy, bCopy := a.DeepCopy(), b.DeepCopy()
    normalizeNodeMeta(aCopy)
    normalizeNodeMeta(bCopy)
	return equality.Semantic.DeepEqual(aCopy, bCopy)
}
```
or even better:
```
func nodeMetadataEqual(a, b metav1.ObjectMeta) bool {
	return equality.Semantic.DeepEqual(a.Labels, b.Labels)
}
```
but we can leave the second variant for follow up. I will open issue.